packages/shell-api/src/collection.ts

```diff
@@ -1418,4 +1425,92 @@ export default class Collection extends ShellApiClass {
       }
     );
   }
+
+  @returnsPromise
+  async getShardVersion(): Promise<any> {
+    this._emitCollectionApiCall('getShardVersion', {});
+    return await this._mongo._serviceProvider.runCommand(
+      ADMIN_DB,
+      {
+        getShardVersion: `${this._database._name}.${this._name}`
+      }
+    );
+  }
+
+  @returnsPromise
+  async getShardDistribution(): Promise<any> {
+    this._emitCollectionApiCall('getShardDistribution', {});
+
+    const result = {} as any;
+    const config = this._mongo.getDB('config');
+
+    const isSharded = !!(await config.getCollection('collections').countDocuments({
+      _id: `${this._database._name}.${this._name}`,
+      dropped: false
+    }));
+    if (!isSharded) {
+      throw new MongoshInvalidInputError(`Collection ${this._name} is not sharded`);
+    }
+
+    const collStats = await (await this.aggregate({ '$collStats': { storageStats: {} } })).toArray();
+
+    const totals = { numChunks: 0, size: 0, count: 0 };
+    const conciseShardsStats = [];
+
+    for (const extShardStats of collStats) {
+      // Extract and store only the relevant subset of the stats for this shard
+      const shardStats = {
+        shardId: extShardStats.shard,
+        host: (await config.getCollection('shards').findOne({ _id: extShardStats.shard })).host,
+        size: extShardStats.storageStats.size,
+        count: extShardStats.storageStats.count,
+        numChunks:
+          await config.getCollection('chunks').countDocuments({ ns: extShardStats.ns, shard: extShardStats.shard }),
```

Comment:
These `await`s could also be pararellized here, considering that it’s used in a loop?

Reply:
Just pushed the change, let me know if you think it looks ok!